app/views/maintenances/_form.html.haml

```diff
@@ -1,7 +1,7 @@
 = simple_form_for @maintenance do |f|
   = f.input :name, as: :string
   = f.input :mileage, as: :integer
-  = f.input :warranty_date, as: :string, input_html: {class: 'datepicker'}
+  = f.input :warranty_date, as: :string, input_html: { class: 'datepicker', value: localize(f.object.warranty_date, format: '%d-%m-%Y')}
```

Comment:
Good idea!
You can shorten it to ``` l(f.object.warranty_date) ``` or ``` l f.object.warranty_date ```
l = localize
The default format for PL is set : %d-%m-%Y. Better don't set it here. Other languages won't be formated correctly.

Reply:
I think need to be:
  = f.input :warranty_date, as: :string, input_html: { class: 'datepicker', value: (l(f.object.warranty_date, format: '%d-%m-%Y') if f.object.warranty_date.present?) }

Where then should be format included?